### ISSUE TITLE
Memory Required for Installation Pimgento2API through Composer

### ISSUE THREAD

**Issue by vbidou** [Agence-DnD/PIMGento-2-API#49]
Dear Team, I may need your help here, just to verify what are the memory requirements in order to install PimGento2-API.
Installation is failing. I receive an error message that it cannot allocate memory.
Do you point my webhosting plan ? 
What Minimum Memory allocation is required for installation with composer?
Thanks in advance.
Best regards,

###########
technos1@es48 [~/public_html]# composer require agencednd/module-pimgento-2-api
Using version ^100.1 for agencednd/module-pimgento-2-api
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
mmap() failed: [12] Cannot allocate memory
mmap() failed: [12] Cannot allocate memory
mmap() failed: [12] Cannot allocate memory
Fatal error: Out of memory (allocated 1040195584) (tried to allocate 33554440 bytes) in phar:///usr/local/bin/c   omposer.phar/src/Composer/DependencyResolver/RuleSet.php on line 77

**Comment by vbidou** [comment]
Merci Rémi !
My PHP.ini memory is set on 768Mo so indeed it should be enough in theory.
I am installing the module on an existing Magento instance.

Tomorrow i'll try few things in order to try fixing this. 
I will document the outcomes here, but in the meantime, any idea is welcome :-)

**Comment by JelleGroenendal** [comment]
Do a composer require on your local machine upload the composer lock file to your server and do a composer install. Composer install is allot less memory intensive compared to composer require.

**Comment by vbidou** [comment]
Dear JelleGroenendal, thanks for your feedback. Really appreciated.
Attempting to use composer Install instead of Composer Require generated still some errors.
I need to train myself in the proper use of composer, because I am still a beginner.

I managed to get a mapping how composer install work ([here](https://adamcod.es/2013/03/07/composer-install-vs-composer-update.html))
When I run Composer Install, from /public_html folder where is installed Magento, it checks if I have a composer.lock 
It's the case, because I have the legacy composer.lock file on this folder.
Then it checks if composer.lock is updated.

Here is my current problem. I receive an error message that it's not updated and it kills the installation process.
**Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run update to update them.
Nothing to install or update
Generating autoload files**

As I don't know how to update composer.lock with the latest changes in composer.json, I will continue to train myself and post on my progress. 
If you have any advise on how to point it out, you are very welcome.

**Comment by vbidou** [comment]
Dear Friends, 
I ow you some feedback where I currently stand on this one.
Discussing with the Guru's of my Hosting, they identified that the problem of Memory allocation was associated to the php version which I use (it was 7.1). They advised me to move to php72 and indeed it resolved my general problem which I had with composer. I moved to php7.2.13

But my problem now is that pimgento-2-api refuse to install the package with this version.
If you have a trick, please let me know. I will inform the guru accordingly to see what can be done under version 7.1.0

Here are the errors I get:
**technos1@es48 [~/public_html]# composer require agencednd/module-pimgento-2-api
  [InvalidArgumentException]
  Package agencednd/module-pimgento-2-api at version  has a PHP requirement incompatible with
  your PHP version (7.2.13)**

**Comment by Dnd-Gimix** [comment]
Hello @vbidou ,

Any news on the subject ?

Regards,

**Comment by Dnd-Gimix** [comment]
Hello @vbidou ,

Is PHP 7.2 compatibility OK for you ?

Regards,

**Comment by vbidou** [comment]
Dear Rémi,

Sorry for the late feedback since I've been 2 weeks in Russia and just returned.
The attempt to install 100.1.4 leaded to crash completely Magento, because of wrong composer manipulations. I had to restart my testing environment. Hence I reinstalled Magento this time with version 2.3 and I'll do a test within 2 weeks. I'll surely keep you posted with the outcomes. Sorry for the delay.

Best regards
